### PR TITLE
Display units (unit conversion refactor)

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request compiles all the efforts in the unit conversion refactor to support display units and will remain in draft mode until all those individual efforts are merged into the development branch.  Until then, this can serve as a thread to track items that need to be done/tested before eventual merge into main.

Incremental PRs:
* #2048
* #2128
* #2130
* #2131
* #2132
* #2148
* #2195
* #2215
* #2227
* #2216
* #2235

Bugs that need fixing upstream before merge:
* #2125
* #2126

Needs testing:
- [x] Update pin once glue 1.9 issues for CI are fixed
- [x] Fix all CI tests (#2132)
- [ ] Cubeviz components: https://github.com/spacetelescope/jdaviz/pull/2128#pullrequestreview-1373114346 (although likely deferring implementation outside specviz for initial release)
- [x] Cubeviz cube viewer axes labels seem to have reverted (but not caught by CI - should have regression tests added)
- [x] test component selection downstream in lcviz, see https://github.com/spacetelescope/lcviz/pull/24#issuecomment-1570639772
- [ ] Mosviz cannot load unit conversion: https://github.com/spacetelescope/jdaviz/pull/2128#issuecomment-1497896224 (although likely deferring implementation outside specviz for initial release)

Other TODOs:

- [ ] any other attributes (meta, wcs, etc) in `core.helpers.ConfigHelper._get_data._handle_display_units`
- [x] ~2d-spectrum-viewer (specviz2d, mosviz) support or disable plugin~ (deferring support)
- [x] ~image viewer support (in mouseover) for cubeviz~ (deferring support)
- [ ] consider renaming plugin (and update docs accordingly)

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
We need to figure out the problem caused by glue-core v1.9.0 first, see #2136

File: CHANGES.rst [hunk 1]
```diff
@@ -22,6 +22,8 @@ Specviz2d
 API Changes
 -----------
 
+- Subset Plugin now respects the chosen display unit after using Unit Conversion. [#2195]
+
```
Comment (kecnry) [inline CHANGES.rst:26]: probably should be removed from the changelog so there is a single entry when merging this into main

**Review by rosteen — APPROVED** [review]
Approved, a few things don't work perfectly in line lists but we were going to defer that to a larger overhaul of line lists.

File: CHANGES.rst (hunk 1 above)
Comment (kecnry) [reply CHANGES.rst:36]: did this get removed (@rosteen)?

Comment (rosteen) [reply CHANGES.rst:36]: Ugh, I did it in my previous squash attempt but not when I redid it. I'll open a PR.

**Pull request merged**